src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -16,6 +18,13 @@ namespace ModelContextProtocol.Shared;
 /// </summary>
 internal sealed class McpSession : IDisposable
 {
+    private static readonly ActivitySource s_activitySource = new("ModelContextProtocol");
+    private static readonly Meter s_meter = new("ModelContextProtocol");
```

Comment:
Given that we have ILoggerFactory support, shouldn't we also leverage IMeterFacotry? @JamesNK has a lot of experience with this.

Reply:
From asking about this previously in the context of M.E.AI, I got the impression that IMeterFactory was much less important than ILoggerFactory, and really only there for testing reasons. Is it more than that? We don't use an IMeterFactory in M.E.AI, for example.

Reply:
It's nice not to have to disable test parallelization.

ASP.NET Core uses it extensively, and it's also supported by SocketsHttpHandler. I'm not sure what would make M.E.AI special. I figured in the places where we aren't using IMeterFactory yet, it was due to it mostly due to stuff written before .NET 8, and we hadn't decided was worth the time and effort to add pluggability yet. But generally, why put all the effort to creating IMeterFactory and using it in a bunch of places not to take advantage of it here? It's not like I'm asking for us to build some abstraction from scratch.

https://learn.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.meterfactory

Reply:
IMeterFactory is great when you always have DI involved, e.g. server. It's more difficult in client libraries because they're usually created without DI being involved. But it's simple enough to add a property like SocketsHttpHandler did.

But I doubt many people in the real world are unit testing metrics.

Reply:
I see it as an implementation detail and is really about how you want to write tests for the repo. It has negligible affect on how consumers will use the library/telemetry.